Release 25082021

TEdwardJ opened the pull request.

Greengrer committed:
Attempted cors configuration in security.

Greengrer committed:
Working cors configuration in security configuration

TEdwardJ committed:
Rest Password Recovery Controller:
 - Rest Controller for password recovery functionality is added;
 - tests are added for the controller;
 - PasswordRecoveryService.generateToken() method added in order to control the returned value during tests execution;

TEdwardJ committed:
Rest Password Recovery Controller:
 - RestPasswordRecoveryController renamed to PasswordRecoveryController;

TEdwardJ committed:
Merge branch 'master' of https://git.heroku.com/json-exchange-implementation into password_recovery
	# Please enter a commit message to explain why this merge is necessary,

Greengrer committed:
Removed "theme" from contact us, changed captcha secret

Greengrer committed:
Changed Discogs callback.url property

Greengrer committed:
Merge pull request #13 from littlepony99/contact_us_standardisation_17082021

Contact us standardisation 17082021

Greengrer committed:
Changed /login /token

Greengrer committed:
Added ServerException, changed contact us and it's responses according to API.

Greengrer committed:
Changed password-recovery to return 200 instead of 201, changed test to not include result-codes.

Greengrer committed:
Changed format of link with token for password recovery to restful.

Greengrer committed:
Changed format of link with token for password recovery to match api.

TEdwardJ committed:
Profile controller moved to rest:
 - change discogs user`s name functionality;
 - change password functionality;
 - UserDao has got 2 new methods for password changing and for discogsUserName editing;
  - UserService has got 2 new methods for password changing and for discogsUserName editing;

Greengrer committed:
Changed уÑemail-confirmation and sign-up according to API.

Greengrer committed:
Merge pull request #14 from littlepony99/new_api_fix_controllers

New api fix controllers

Greengrer committed:
Merge branch 'profile_controller' of https://github.com/littlepony99/wax-deals into profile_controller

Greengrer committed:
Changed profile controller test to use @beforeEach.

TEdwardJ committed:
Profile controller moved to rest:
 - Order of parameters in the DefaultUserService.changeDiscogsUserName() method in order to make the order the same as in JdbcUserDao.changeDiscogsUserName();

TEdwardJ committed:
Profile controller:
 - profile management logic moved to ProfileManagementService from ProfileController;
 - JwtTokenProvider moved to impl package;
 - ProfileController test fix;

TEdwardJ committed:
Change Password functionality:
 - controller should receive old password along with new password and new password confirmation;

TEdwardJ committed:
Password Recovery functionality:
 - rest-styled "/password-recovery/{recoveryToken}" endpoint added to the password recovery controller;;

TEdwardJ committed:
Password Recovery test:
 - elimination of unnecessary code;

TEdwardJ committed:
Email Confirmation new endpoint:
 - confirmation token is expected to be sent in PUT request body;

TEdwardJ committed:
Possibility to edit user`s email:
 - Besides of possibility to edit the discogsUserName attribute, possibility to edit user`s email added;

TEdwardJ committed:
OneVinylOfferDto Mapping changes:
 - id field is added into OneVinylOfferDto;

TEdwardJ committed:
OneVinylOfferDto Mapping changes:
 - id field mapping issue fix - ambiguity for id fields eliminated;

TEdwardJ committed:
Profile Management changes:
 - Http Code 400 is returned in case the user specified wrong old password during password change process;
 - new JWT token is returned in case the user changed their email;

TEdwardJ committed:
Profile Controller PR review:
 - elimination of commented code;
 - changes in the variable naming;
 - elimination of redundant empty lines;;

TEdwardJ committed:
Profile Controller PR review:
 - JwtAuthenticationException constructor Throwable-typed parameter has been renamed;

TEdwardJ committed:
Merge pull request #15 from littlepony99/profile_controller

Profile controller

TEdwardJ committed:
Release 25082021 PR review:
 - unused imports elimination;

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcUserDao.java:
```diff
@@ -54,29 +62,29 @@ public long add(User user) {
     public void delete(User user) {
-        MapSqlParameterSource sqlParameterSource = new MapSqlParameterSource();
-        sqlParameterSource.addValue("email", user.getEmail());
+        MapSqlParameterSource sqlParameterSource = new MapSqlParameterSource()
```
jdbcTemplate.update(DELETE, new MapSqlParameterSource().addValue("email", user.getEmail()));

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcUserDao.java:
```diff
@@ -30,6 +30,14 @@
+
     private static final String UPDATE_USER_STATUS = "UPDATE users SET status = true WHERE id = :id";
 
     private static final ResultSetExtractor<User> RESULT_SET_EXTRACTOR = new UserResultSetExtractor();
```
Please rename namedParameterJdbcTemplate to jdbcTemplate

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcUserDao.java:
```diff
@@ -54,29 +62,29 @@ public long add(User user) {
 
```
From:
Long id = Long.parseLong(keyHolder.getKeys().get("id").toString());
        return id;

To
return  Long.parseLong(keyHolder.getKeys().get("id").toString());

Boyscout rule. It could be not yours code but this is our codebase . We should do our best to clean it up every time we visit a class. We should review whole class and clean it up if we see bad code

TEdwardJ replied:
This change is already made in another branch and merged into develop

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcUserDao.java:
```diff
@@ -54,29 +62,29 @@ public long add(User user) {
     }
 
     @Override
     public Optional<User> findByEmail(String email) {
```
>  Please add static import of Optional and rewrite as follows:
   @Override
    public Optional<User> findByEmail(String email) {
        MapSqlParameterSource sqlParameterSource = new MapSqlParameterSource().addValue("email", email);
        return ofNullable(jdbcTemplate.query(FIND_BY_EMAIL, sqlParameterSource, RESULT_SET_EXTRACTOR));
    }

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcUserDao.java:
```diff
@@ -85,8 +93,8 @@ public void update(String email, User user) {
 
     @Override
     public Optional<User> findById(long id) {
```
>     @Override
    public Optional<User> findById(long id) {
        MapSqlParameterSource sqlParameterSource = new MapSqlParameterSource().addValue("id", id);
        return ofNullable(jdbcTemplate.query(FIND_BY_ID, sqlParameterSource, RESULT_SET_EXTRACTOR));
    }

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcUserDao.java:
```diff
@@ -95,9 +103,26 @@ public void update(String email, User user) {
 
     @Override
     public void setUserStatusTrue(long id) {
```
>     @Override
    public void setUserStatusTrue(long id) {
        MapSqlParameterSource sqlParameterSource = new MapSqlParameterSource().addValue("id", id);
        jdbcTemplate.update(UPDATE_USER_STATUS, sqlParameterSource);
    }

TEdwardJ replied:
Artem, could you please explain, what the issue is supposed to be here?

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcUserDao.java:
```diff
@@ -95,9 +103,26 @@ public void update(String email, User user) {
     }
 
+    @Override
+    public void changeProfile(User user, String email, String discogsUserName) {
```
>    @Override
    public void changeProfile(User user, String email, String discogsUserName) {
        MapSqlParameterSource sqlParameterSource = new MapSqlParameterSource()
                .addValue("new_email", email)
                .addValue("discogs_user_name", discogsUserName)
                .addValue("email", user.getEmail());
        jdbcTemplate.update(UPDATE_PROFILE_FIELDS, sqlParameterSource);
    }

TEdwardJ replied:
Artem, could you please explain, what the issue is supposed to be here?

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcUserDao.java:
```diff
@@ -95,9 +103,26 @@ public void update(String email, User user) {
+    }
+
+    @Override
+    public void changeUserPassword(User user) {
```
>    @Override
    public void changeUserPassword(User user) {
        MapSqlParameterSource sqlParameterSource = new MapSqlParameterSource()
                .addValue("email", user.getEmail())
                .addValue("password", user.getPassword());
        jdbcTemplate.update(UPDATE_PASSWORD, sqlParameterSource);
    }

TEdwardJ replied:
Artem, could you please explain, what the issue is supposed to be here?

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/exception/JwtAuthenticationException.java:
```diff
@@ -0,0 +1,13 @@
+import org.springframework.security.core.AuthenticationException;
+
+public class JwtAuthenticationException extends AuthenticationException {
+    public JwtAuthenticationException(String message, Throwable throwable) {
```
+ extra line break here

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/exception/ServerException.java:
```diff
@@ -0,0 +1,9 @@
+    public ServerException(final String message) {
+        super(message);
+    }
+
```
remove line break here

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/exception/entity/EmailConfirmationErrors.java:
```diff
@@ -14,7 +14,8 @@
     CAN_NOT_ADD_LINK_FOR_EMAIL("Sorry, can't create link to confirm that email."),
-    CAN_NOT_CREATE_LINK_TRY_AGAIN("Sorry, error on our side, could you try one more time? Contact us if it doesn't work");
+    CAN_NOT_CREATE_LINK_TRY_AGAIN("Sorry, error on our side, could you try one more time? Contact us if it doesn't work"),
+    CAN_NOT_SEND_EMAIL(" Sorry, we couldn't send you a link to confirm your email. Please, try again or contact us.");
```
Why white space between " and Sorry ?

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/security/SecurityConfiguration.java:
```diff
@@ -69,15 +68,17 @@ protected void configure(HttpSecurity http) throws Exception {
                 .logout().addLogoutHandler(logoutHandlerService).logoutSuccessUrl("/successlogout");
     }
 
-/*    @Bean
```
YAGNI

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/DefaultProfileManagementService.java:
```diff
@@ -0,0 +1,64 @@
+package com.vinylteam.vinyl.service;
+
+import com.vinylteam.vinyl.dao.jdbc.extractor.DefaultUserMapper;
```
not used

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/DefaultProfileManagementService.java:
```diff
@@ -0,0 +1,64 @@
+        User user = (User) request.getAttribute("userEntity");
+        String oldEmail = user.getEmail();
+
+        User updatedUser = userService.changeProfile(user, userProfileInfo.getEmail(), userProfileInfo.getDiscogsUserName());
```
Variable declaration does not make sense

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/DefaultProfileManagementService.java:
```diff
@@ -0,0 +1,64 @@
+        User updatedUser = userService.changeProfile(user, userProfileInfo.getEmail(), userProfileInfo.getDiscogsUserName());
+        var response = ControllerResponseUtils.getResponseWithMessage("Your email and/or discogs username have been changed.");
+        if (!user.getEmail().equals(oldEmail)) {
+            response.setToken(jwtService
```
one line
response.setToken(jwtService.createToken(user.getEmail(), userMapper.mapToDto(user).getAuthorities()));

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/DefaultProfileManagementService.java:
```diff
@@ -0,0 +1,64 @@
+import org.springframework.stereotype.Service;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.Optional;
```
not used

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/UserService.java:
```diff
@@ -1,28 +1,32 @@
+
     void signInCheck(UserInfoRequest userProfileInfo);
 
     User editProfile(UserInfoRequest userProfileInfo,
```
one line
User editProfile(UserInfoRequest userProfileInfo, User user) throws ServerException;

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultEmailConfirmationService.java:
```diff
@@ -20,7 +21,7 @@
 public class DefaultEmailConfirmationService implements EmailConfirmationService {
```
Boyscout rule:

    @Override
    public UUID generateConfirmationToken() {
        UUID uuid = UUID.randomUUID();
        return uuid;
    }

remove variable declaration and just return UUID.randomUUID() please

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultPasswordRecoveryService.java:
```diff
@@ -24,7 +25,7 @@
 @RequiredArgsConstructor
 public class DefaultPasswordRecoveryService implements PasswordRecoveryService {
-    private static final String RECOVERY_MESSAGE = "Hello, to change your password, follow this link: \n{applicationLink}/recoveryPassword/newPassword?token=";
+    private static final String RECOVERY_MESSAGE = "Hello, to change your password, follow this link: \n{applicationLink}/new-password/";
```
please add extra line break

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultPasswordRecoveryService.java:
```diff
@@ -74,13 +73,19 @@ public void checkToken(String token) {
 
+    @Override
+    public UUID generateToken() {
+        UUID uuid = UUID.randomUUID();
```
Variable declaration does not make sense

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultUserPostService.java:
```diff
@@ -30,45 +32,40 @@
 
-    String createContactUsMessage(String recipient, String subject, String mailBody) {
+    String createContactUsMessage(String recipient, String mailBody) {
         return "MailFrom: " +
```
one line
return "MailFrom: " + recipient + System.lineSeparator() + "Message: " + mailBody;

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultUserService.java:
```diff
@@ -12,14 +12,13 @@
-import java.util.UUID;
 
 @Slf4j
 @RequiredArgsConstructor
```
please remove 
//securityService.emailFormatCheck(email); line at all
Boyscout rule + YAGNI

plus
line 45+46 should be one line 
User userToAdd = securityService.createUserWithHashedPassword(email, password.toCharArray());

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultUserService.java:
```diff
@@ -141,7 +174,7 @@ public void signInCheck(UserInfoRequest userProfileInfo) {
 
     @Override
     public User editProfile(UserInfoRequest userProfileInfo,
```
one line
public User editProfile(UserInfoRequest userProfileInfo, User user) throws ServerException {

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/util/ControllerResponseUtils.java:
```diff
@@ -15,21 +15,21 @@
-        response.setResultCode(code);
         return response;
     }
 
```
line 43-43
please make it one line
return Map.of("user", userMapper.mapUserToDto(user), "token", token);

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/controller/ContactUsController.java:
```diff
@@ -2,55 +2,44 @@
     private final UserPostService userPostService;
 
     @Value("${project.mail}")
     private String projectMail;
```
not used

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/controller/ProfileController.java:
```diff
@@ -1,84 +1,60 @@
-        WebUtils.setUserAttributes(user, model);
-        return "editProfile";
+    public ResponseEntity<UserSecurityResponse> submitProfileChanges(HttpServletRequest request, @RequestBody UserInfoRequest userProfileInfo) {
+        var response = profileService.changeProfileAndReturnUser(request, userProfileInfo);
```
variable declaration does not make sense

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/controller/ProfileController.java:
```diff
@@ -1,84 +1,60 @@
-            return new ModelAndView("redirect:/signIn");
-        }
+    public ResponseEntity<ChangePasswordResponse> changePassword(HttpServletRequest request, @RequestBody UserInfoRequest userProfileInfo) {
+        ChangePasswordResponse response = profileService.changeProfilePassword(request, userProfileInfo);
```
variable declaration does not make sense

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/controller/ProfileController.java:
```diff
@@ -1,84 +1,60 @@
 package com.vinylteam.vinyl.web.controller;
 
```
A ot of unused imports
Please clean up

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/controller/SignupController.java:
```diff
@@ -3,6 +3,7 @@
 import com.vinylteam.vinyl.dao.jdbc.extractor.UserMapper;
```
A lot of unused imports
Please clean up

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/controller/SignupController.java:
```diff
@@ -19,56 +20,58 @@
 
-    private Map<String, String> getStatusInfoMap(String code, String s) {
+    private Map<String, String> getMessageMap(String s) {
         return Map.of(
```
rename s > message
and make one line
return Map.of("message", message);

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/controller/SignupController.java:
```diff
@@ -19,56 +20,58 @@
 @Slf4j
 @RequiredArgsConstructor
-@CrossOrigin(origins = {"http://localhost:3000", "https://react-wax-deals.herokuapp.com"})
 public class SignupController {
```
Why do we need to return Map<String, Object> and not Map<String, String> ?
As far as I am concerned you always return Map<String, String> but have to wrap it due to the methods return type
Why we cannot just return Map<String, String> from controller methods?

Feel free to call me to discuss

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/dto/ChangePasswordResponse.java:
```diff
@@ -0,0 +1,14 @@
+@Data
+@RequiredArgsConstructor
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class ChangePasswordResponse {
```
add line break

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/handler/ApiExceptionHandler.java:
```diff
@@ -1,22 +1,44 @@
 
+    @ExceptionHandler(value = {BadCredentialsException.class})
+    public ResponseEntity<Object> handleBadCredentialsExceptionHandler(BadCredentialsException exception) {
+        ApiExceptionDto apiExceptionDto = new ApiExceptionDto(
```
one line
return ResponseEntity.status(BAD_REQUEST).body(new ApiExceptionDto(exception.getMessage()));

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/handler/ApiExceptionHandler.java:
```diff
@@ -1,22 +1,44 @@
+
     @ExceptionHandler(value = {RuntimeException.class})
-    public ResponseEntity<Object> handleApiExceptionHandler(RuntimeException exception) {
+    public ResponseEntity<Object> handleRuntimeExceptionApiExceptionHandler(RuntimeException exception) {
```
return ResponseEntity.status(FORBIDDEN).body(new ApiExceptionDto(exception.getMessage()));

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/handler/ApiExceptionHandler.java:
```diff
@@ -1,22 +1,44 @@
+    }
+
+    @ExceptionHandler(value = {ServerException.class})
+    public ResponseEntity<Object> handleServerExceptionApiExceptionHandler(ServerException exception) {
```
return ResponseEntity.status(INTERNAL_SERVER_ERROR).body(new ApiExceptionDto(exception.getMessage()));

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/handler/ApiExceptionHandler.java:
```diff
@@ -1,22 +1,44 @@
+
+    @ExceptionHandler(value = {ForbiddenException.class})
+    public ResponseEntity<Object> handleForbiddenExceptionApiExceptionHandler(ForbiddenException exception) {
+        ApiExceptionDto apiExceptionDto = new ApiExceptionDto(
```
return ResponseEntity.status(BAD_REQUEST).body(new ApiExceptionDto(exception.getMessage()));

TEdwardJ committed:
Release 25082021 PR review:
 - unused imports elimination;
 - commented lines of code were removed;
 - variables renaming;

TEdwardJ committed:
JdbcUserDao:
 - SqlParameterSource as a dedicated variable in all corresponding methods;

TEdwardJ committed:
JdbcUserDao:
 - SqlParameterSource as a dedicated variable in all corresponding methods;

TEdwardJ committed:
PR release_25082021->develop conflict resolution

TEdwardJ committed:
PR release_25082021:
 - elimination of the obsolete variable;
